server/schedule/checker_controller.go

```diff
@@ -52,8 +55,8 @@ func (c *CheckerController) CheckRegion(region *core.RegionInfo) (bool, []*opera
 	// Don't check isRaftLearnerEnabled cause it maybe disable learner feature but there are still some learners to promote.
 	opController := c.opController
 	checkerIsBusy := true
-	if c.cluster.IsPlacementRulesEnabled() {
-		if opController.OperatorCount(operator.OpReplica) < c.cluster.GetReplicaScheduleLimit() {
+	if c.opts.IsPlacementRulesEnabled() {
```

Comment:
I think it's better to use one of `c.opts` or `c.GetOpts()` for all code.

Reply:
em, it is a [shortcut](https://github.com/tikv/pd/pull/2917/files/b6bebcb89d8286f3d5848328af6fa3e6a25b961f#diff-d65128753ed35aa8bf31ad063ce2af76R30) I used to make code cleaner. (`c.opts.IsPlacementRulesEnabled` instead of `c.cluster.GetOpts().IsPlacementRulesEnabled`)